### ISSUE TITLE
start offsets

### ISSUE THREAD

**Issue by JohnPaulHarold** [alemangui/pizzicato#7]
Currently, you can make a `Sound`, and play it. But what if I want to make the `Sound`, and play it 10 seconds later, or at a specific `context.currentTime`?

What's stopping Pz from doing this, if anything?

**Comment by alemangui** [comment]
Nothing really, it's just a use case that hasn't come up. I would be OK with implementing this. Perhaps as an optional parameter to the ```play```function.

**Comment by JohnPaulHarold** [comment]
OK, have got something almost ready to go with this, but I'm getting failing tests. I pulled down a zip of master and ran gulp tests in that, and that's also reporting the same failing tests.

```
Firefox 47.0.0 (Windows 10 0.0.0) Sound actions should trigger 'end' when ended FAILED
        Expected spy endCallback to have been called.
        sound</<@C:/Users/John-Paul Harold/Code/pizzicato/tests/Sound.test.js:237:6
Firefox 47.0.0 (Windows 10 0.0.0) Sound actions should stop automatically when file sound is over FAILED
        Expected true to be false.
        click</<@C:/Users/John-Paul Harold/Code/pizzicato/tests/Sound.test.js:261:6
```

@alemangui  Do you see tests failing also? I can raise this as a separate issue if required.

**Comment by alemangui** [comment]
I have been taking a look at this as well. I have an API prepared that would allow to pass 2 parameters two the ```play``` function:

```sound.play(when, offset);```

So calling ```sound.play(2, 4);``` will play the sound after two seconds starting from 00:04.

Do you have any remarks on this API?

There is one main issue that has prevented me from merging this. Which was described by you in #1: the offset is off for tracks loaded from external files that loop. I will open a separate issue for this.

The second issue has to do with testing. In TravisCI, sometimes I get a failing test ([link here](https://travis-ci.org/alemangui/pizzicato/builds/138569619)) that is related to the loading of a wav file. My guess is that it's linked to the timeout time of that particular test, but I have not been able to reproduce it locally.

I will enter both issues separately, and maybe a case can be made for merging the code for the ```when``` and ```offset``` parameters and deal with the main issue separately.

**Comment by JohnPaulHarold** [comment]
I've pushed my local branch if you wish to compare against it, but the only real change is that I added the duration param for completeness.

So, the idea on the first param is that it will always add the context.currentTime to the user desired when? I think I like that, good choice :-) With mine, I'd just assumed the user would be taking care of that, but I think your's is a nice touch. Right now, I can't think of any downsides to this.
